lib_staging.ks

```diff
@@ -1,51 +1,96 @@
-// Add all fuels you want to check in to this list (e.g. Karbonite if you use the mod),
-// but SolidFuel is intentionaly left out and Oxidizer is unimportant
-global stagingFuelList	is list("LiquidFuel").
-global stagingNumber	is -1.
-global stagingEngines	is list().
-global stagingTanks		is list().
-
-// to be called whenever current stage changes to prepare data for quicker test.
-//
-// prepares a list of engines that are in current stage but not in next stage.
-// counts fuel in each and decides which to add to the "flameout watch list"
-// that drives the staging behavior.
-//
-// HOW IT WORKS
-// ------------
-// for most parts, :STAGE is the stage number they separate in; for engines,
-// this is the number they activate in.
-// e:parent:stage is different works for any engine that is not attached
-// directly to another engine. boosters are engines and therefore this fails
-// for booster-on-booster arrangements; all will stage at once even if they're
-// not all depleted.
+// Library for staging logic and deltaV calculation
+// ================================================
+// Asparagus and designs that throw off empty tanks were considered.
+// Note that engines attached to tanks that get empty will be staged
+// (even if not technically flamed out - that is something that KER and MechJeb do not consider).
+
+// LOGIC: Stage if either availableThrust = 0 (separator-only, fairing-only stage)
+// or all engines that are to be separated by staging flame out
+// or all tanks to be separated that were not empty are empty now
+
+// list of all consumed fuels (for deltaV; add e.g. Karbonite and/or MonoPropellant if using such mods)
+global stagingConsumed is list("Oxidizer","SolidFuel", "LiquidFuel").
+// list of fuels for empty-tank identification (for dual-fuel tanks use only one of the fuels)
+global stagingTankFuels is list("LiquidFuel"). //Oxidizer and SolidFuel intentionally not included
+
```

Comment:
Maybe `SolidFuel` should be in the list and we could use `stagingDecoupledIn` for tanks as well. That should not throw off tank+booster combo if booster is still lit. I will test that when I have the time.